src/components/Sidebar/cart/CartSection.test.js

```diff
@@ -38,4 +38,15 @@ describe('CartSection', () => {
 
     expect(wrapper.find(Section).prop('leftHeaderContent')).toBe('schedule, schedule');
   });
+
+  it('renders modal correctly', () => {
+    const classes = mockStyles(styles);
+    const testSection = { schedule: [{}], subjectId: 'EECS', courseId: '111-0' };
+    const wrapper = shallow(
+      <UnstyledCartSection section={testSection} classes={classes} />,
+    );
+    wrapper.instance().toggleDialog();
```

Comment:
One thing we should *never* do in unit tests - manipulate "private" methods or state of the component we are testing. Why? Because the consumers of our component will never be able to, so why should our tests? We want our tests to replicate a case of our component being used, and if no consumer will be able to, for example, call the `toggleDialog` method, why should the test? This breaks our abstraction of implementation details, so we should never do this. In most cases, if you're using `wrapper.instance()` or directly touching component state from a test, that's a red flag.

In this case, think about how a user would toggle this dialog. They would click the section, right? Which, as far as our component is concerned, is calling the `Section`'s `onClick`. To simulate that, we can do:
```
wrapper.find(Section).simulate('click');
```
Then, we can confirm that the modal is now open by checking to see if the `ClassModalContainer`'s `showDialog` prop is set to `true` (because, again, as far as our component here is concerned, if that prop is set to true, the `ClassModalContainer` component will take care of opening itself):
```
expect(wrapper.find(ClassModalContainer).prop('showDialog')).toBe(true);
```

Oh, and we should probably refactor `ClassModalContainer` to be `ClassDialogContainer`... but perhaps for a different ticket.